server/channels/app/notification.go

```diff
@@ -663,6 +608,175 @@ func (a *App) SendNotifications(c request.CTX, post *model.Post, team *model.Tea
 	return mentionedUsersList, nil
 }
 
+func (a *App) RemoveNotifications(c request.CTX, post *model.Post, channel *model.Channel, team *model.Team) error {
+	isCRTAllowed := *a.Config().ServiceSettings.CollapsedThreads != model.CollapsedThreadsDisabled
+
+	// CRT is the main issue in this case as notifications indicator are not updated when accessing threads from the sidebar.
+	if isCRTAllowed && post.RootId != "" {
+		pCh := make(chan store.StoreResult, 1)
+		go func() {
+			props, err := a.Srv().Store().User().GetAllProfilesInChannel(context.Background(), channel.Id, true)
+			pCh <- store.StoreResult{Data: props, NErr: err}
+			close(pCh)
+		}()
+
+		cmnCh := make(chan store.StoreResult, 1)
+		go func() {
+			props, err := a.Srv().Store().Channel().GetAllChannelMembersNotifyPropsForChannel(channel.Id, true)
+			cmnCh <- store.StoreResult{Data: props, NErr: err}
+			close(cmnCh)
+		}()
+
+		var gCh chan store.StoreResult
+		if a.allowGroupMentions(c, post) {
+			gCh = make(chan store.StoreResult, 1)
+			go func() {
+				groupsMap, err := a.getGroupsAllowedForReferenceInChannel(channel, team)
+				gCh <- store.StoreResult{Data: groupsMap, NErr: err}
+				close(gCh)
+			}()
+		}
+
+		result := <-pCh
+		if result.NErr != nil {
+			return result.NErr
+		}
+		profileMap := result.Data.(map[string]*model.User)
+
+		result = <-cmnCh
+		if result.NErr != nil {
+			return result.NErr
+		}
+		channelMemberNotifyPropsMap := result.Data.(map[string]model.StringMap)
+
+		groups := make(map[string]*model.Group)
+		if gCh != nil {
+			result = <-gCh
+			if result.NErr != nil {
+				return result.NErr
+			}
+			groups = result.Data.(map[string]*model.Group)
+		}
+
+		mentions, _ := a.getExplicitMentionsAndKeywords(c, post, channel, profileMap, groups, channelMemberNotifyPropsMap, nil)
```

Comment:
`getExplicitMentionsAndKeywords` does not handle group mentions. Why are they not important in this context?

Reply:
They are important, just not being handled as part of this PR. I think its because the original ticket was specifically for reply inside a thread. I'll update this to handle user groups.